crates/sui-core/src/authority_server.rs

```diff
@@ -415,16 +417,21 @@ impl ValidatorService {
             return Err(tonic::Status::from(SuiError::ValidatorHaltedAtEpochEnd));
         }
 
-        // 4) If it's a shared object transaction and requires consensus, we need to do so.
-        // This will wait until either timeout or we have heard back from consensus.
-        if is_consensus_tx && !state.consensus_message_processed(&certificate)? {
+        // 4) All certificates are sent to consensus (at least by some authorities)
+        // For shared objects this will wait until either timeout or we have heard back from consensus.
+        // For owned objects this will return without waiting for certificate to be sequenced
+        if !state.consensus_message_processed(&certificate)? {
             // Note that num_inflight_transactions() only include user submitted transactions, and only user txns can be dropped here.
             // This backpressure should not affect system transactions, e.g. for checkpointing.
             if consensus_adapter.num_inflight_transactions() > MAX_PENDING_CONSENSUS_TRANSACTIONS {
                 return Err(tonic::Status::resource_exhausted("Reached {MAX_PENDING_CONSENSUS_TRANSACTIONS} concurrent consensus transactions",
                 ));
             }
-            let _metrics_guard = start_timer(metrics.consensus_latency.clone());
+            let _metrics_guard = if shared_object_tx {
+                Some(metrics.consensus_latency.start_timer())
+            } else {
+                None
+            };
             consensus_adapter.submit(&state.name, &certificate).await?;
```

Comment:
I remember that the plan to eventually submit this to NW in parallel? If so, may want to add a TODO there

Reply:
Added todo in slightly different place(inside consensus_adapter where we do call to worker - this is the part that can be async for owned objects)